examples/network_transport/tcp_sockets_wrapper/ports/lwip/tcp_sockets_wrapper.c

```diff
@@ -35,6 +35,68 @@
 /* configASSERT() using stuff in task.h */
 #include "task.h"
 
+static BaseType_t ConfigureTimeout( Socket_t xSocket,
+                                    uint32_t receiveTimeoutMs,
+                                    uint32_t sendTimeoutMs )
+{
+    BaseType_t xRet = TCP_SOCKETS_ERRNO_NONE;
+    int fcntlFlags = 0;
+    uint32_t realReceiveTimeoutMs = receiveTimeoutMs;
+    uint32_t realSendTimeoutMs = sendTimeoutMs;
+
+    if( ( receiveTimeoutMs == 0 ) &&
+        ( sendTimeoutMs == 0 ) )
+    {
+        fcntlFlags = fcntl( xSocket->xFd, F_GETFL, 0 );
+        if( fcntlFlags < 0 )
+        {
+            LogError( ( "fcntl() failed with errno: %d", errno ) );
+            xRet = TCP_SOCKETS_ERRNO_ERROR;
+        }
+        else
+        {
+            fcntlFlags |= O_NONBLOCK;
+            if( fcntl( xSocket->xFd, F_SETFL, fcntlFlags ) < 0 )
+            {
+                LogError( ( "fcntl() failed with errno: %d", errno ) );
+                xRet = TCP_SOCKETS_ERRNO_ERROR;
+            }
+        }
+    }
+    else
+    {
+        /* If the timeout is not 0, disable the non-blocking flag in socket handler. */
+        fcntlFlags = fcntl( xSocket->xFd, F_GETFL, 0 );
+        if( fcntlFlags < 0 )
+        {
+            LogError( ( "fcntl() failed with errno: %d", errno ) );
+            xRet = TCP_SOCKETS_ERRNO_ERROR;
+        }
+        else
+        {
+            fcntlFlags &= ~O_NONBLOCK;
+            if( fcntl( xSocket->xFd, F_SETFL, fcntlFlags ) < 0 )
+            {
+                LogError( ( "fcntl() failed with errno: %d", errno ) );
+                xRet = TCP_SOCKETS_ERRNO_ERROR;
+            }
+        }
+
+        if( ( receiveTimeoutMs == TCP_SOCKETS_INFINITE_TIMEOUT ) &&
+            ( sendTimeoutMs == TCP_SOCKETS_INFINITE_TIMEOUT ) )
+        {
+            /* Set the timeout to 0 as waiting infinitely. */
+            realReceiveTimeoutMs = 0U;
+            realSendTimeoutMs = 0U;
+        }
+
+        setsockopt( xSocket->xFd, SOL_SOCKET, SO_RCVTIMEO, &realReceiveTimeoutMs, sizeof( realReceiveTimeoutMs ) );
```

Comment:
I see we are not checking setsocket return value anywhere. I think we should check for the result value and fail gracefully.

Reply:
Updated